File: android/app/src/main/java/uk/co/darkerwaters/scorepal/points/Point.java
package uk.co.darkerwaters.scorepal.points;

import android.content.Context;

public interface Point {

    String K_SPEAKING_SPACE = " ";
    String K_SPEAKING_PAUSE = ". ";
    String K_SPEAKING_PAUSE_SLIGHT = ", ";
    String K_SPEAKING_PAUSE_LONG = "… ";//"... ";//""… ";

    int val();

    String displayString(Context context);
    String speakString(Context context);
    String speakString(Context context, int number);
    String speakAllString(Context context);
}

File: multiphone/lib/match/points/point.dart
import 'package:flutter/cupertino.dart';

class PointPair {
  final Point first;
  final Point second;

  PointPair(this.first, this.second);
}

abstract class Point {
  static const String K_SPEAKING_SPACE = " ";
  static const String K_SPEAKING_PAUSE = ". ";
  static const String K_SPEAKING_PAUSE_SLIGHT = ", ";
  static const String K_SPEAKING_PAUSE_LONG = "… "; //"... ";//""… ";

  int val();

  String displayString(BuildContext context);
  String speakString(BuildContext context);
  String speakNumberString(BuildContext context, int number);
  String speakAllString(BuildContext context);
}
